client/src/app/App.js

```diff
@@ -1182,15 +1182,71 @@ export class App extends PureComponent {
     });
   }
 
-  async exportAs(tab) {
+  /**
+   * Exports file to given export type.
+   *
+   * @param {String} options.encoding
+   * @param {String} options.exportPath
+   * @param {String} options.exportType
+   * @param {File} options.originalFile
+   */
+  async exportAsFile(options) {
+    const {
+      encoding,
+      exportType,
+      exportPath,
+      originalFile,
+    } = options;
+
     const {
       globals,
-      tabsProvider
     } = this.props;
 
     const { fileSystem } = globals;
 
+    const contents = await this.tabRef.current.triggerAction('export-as', {
+      fileType: exportType
+    });
+
+    return await fileSystem.writeFile(exportPath, {
```

Comment:
`await` is unnecessary here as we don't catch the error
Cf. https://jakearchibald.com/2017/await-vs-return-vs-return-await/